imapserver/imapmemserver/mailbox.go

```diff
@@ -418,13 +431,40 @@ func (mbox *MailboxView) staticSearchCriteria(criteria *imap.SearchCriteria) {
 }
 
 func (mbox *MailboxView) Store(w *imapserver.FetchWriter, numSet imap.NumSet, flags *imap.StoreFlags, options *imap.StoreOptions) error {
-	mbox.forEach(numSet, func(seqNum uint32, msg *message) {
-		msg.store(flags)
-		mbox.Mailbox.tracker.QueueMessageFlags(seqNum, msg.uid, msg.flagList(), mbox.tracker)
+	var modifiedSeqNums []uint32
+	var modifiedMsgs []*message
+
+	mbox.mutex.Lock()
```

Comment:
Nit: we can use `mbox.forEach` instead of manually locking/unlocking.

Reply:
agreed, updated